aggregations-and-analytics/10/10_moving_average.md

```diff
@@ -0,0 +1,41 @@
+# 01 Moving average
+
+![Twitter Badge](https://img.shields.io/badge/Flink%20Version-1.13%2B-lightgrey)
+
+> :bulb: This example will show how to calculate a moving average in real-time using a `HOP` window.
+
+The source table (`server_logs`) is backed by the [`faker` connector](https://flink-packages.org/packages/flink-faker), which continuously generates rows in memory based on Java Faker expressions.
+
+In one of our previous recipes we've shown how you can [aggregate time series data](../01/01_group_by_window_tvf.md) using `TUMBLE`. 
+To display every 30 seconds the moving average of bidding prices per currency per 1 minute, we will use the built-in `HOP` [function](https://ci.apache.org/projects/flink/flink-docs-stable/docs/dev/table/sql/queries/window-agg/).
+
+The difference between a `HOP` and a `TUMBLE` function is when you stop with the aggregation. A tumbling window groups all the data from the start of the window **until** the end of the window and then starts a new window.
```

Comment:
To me the main difference is that `HOP` (usually) creates overlapping windows, which means that one record (usually) is added to multiple windows. 

I don't understand the "stop with the aggregation". The aggregation is when the window is closed at end of the window (same as TUMBLE). It's just that in the meantime more windows have already been opened for the same key.

"usually", because if the slide interval is larger than the window size, they actually don't overlap and some records are added to no window at all.

Reply:
Would you prefer this?

The difference between a `HOP` and a `TUMBLE` function is that with a `HOP` you can "jump" forward in time. That's why you have to specify both the length of the window and the interval you want to jump forward. When using a `HOP` function, records can be assigned to multiple windows if the interval is smaller than the window length, like in this example. A tumbling window never overlaps and records will only belong to one window.

Reply:
Sounds good.